zeus/optimizer/power_limit.py

```diff
@@ -468,3 +470,150 @@ def _save_profile(self) -> None:
+
+
+# Only import HuggingFace Classes when type checking, to avoid hard dependency on HuggingFace Transformers
+if TYPE_CHECKING:
+    from transformers import (
+        TrainerCallback,
+        TrainingArguments,
+        TrainerState,
+        TrainerControl,
+        PreTrainedModel,
+    )
+
+
+def import_hf_classes():
+    """Attempts to import and return several classes from the `transformers` library.
+
+    This function tries to import the following classes from the `transformers` library:
+    - `TrainerCallback`
+    - `TrainingArguments`
+    - `TrainerState`
+    - `TrainerControl`
+    - `PreTrainedModel`
+
+    Returns:
+        tuple: A tuple containing the imported classes if the `transformers` library is installed.
+        None: If the `transformers` library is not installed, returns None.
+
+    """
+    try:
+        from transformers import (
+            TrainerCallback,
+            TrainingArguments,
+            TrainerState,
+            TrainerControl,
+            PreTrainedModel,
+        )
+
+        return (
+            TrainerCallback,
+            TrainingArguments,
+            TrainerState,
+            TrainerControl,
+            PreTrainedModel,
+        )
+    except ImportError:
+        return None
+
+
+def make_hf(cls: Type[Callback], name: str | None = None) -> Type[TrainerCallback]:
+    """Wrap a `zeus` callback class to be compatible with Hugging Face's Trainer.
+
+    Args:
+        cls (Type[Callback]): The `zeus` callback class to wrap.
+        name (str, optional): The name of the new TrainerCallback class. If None, use `cls.__name__ + "HFWrapper"`.
+
+    """
+    # dynamically import Hugging Face classes, to avoid hard dependency
+    hf_classes = import_hf_classes()
+    if hf_classes is None:
+        raise ImportError(
+            "Hugging Face is not installed. Please install it to use this feature."
+        )
+
+    TrainerCallback, _, _, _, _ = hf_classes  # noqa: N806
+
+    class Wrapper(TrainerCallback):
+        def __init__(self, *args, **kwargs) -> None:
+            self.plo = cls(*args, **kwargs)
+
+        def on_epoch_begin(
+            self,
+            args: TrainingArguments,
+            state: TrainerState,
+            control: TrainerControl,
+            model: PreTrainedModel,
+            **kwargs,
+        ) -> None:
+            self.plo.on_epoch_begin()
+
+        def on_epoch_end(
+            self,
+            args: TrainingArguments,
+            state: TrainerState,
+            control: TrainerControl,
+            model: PreTrainedModel,
+            **kwargs,
+        ) -> None:
+            self.plo.on_epoch_end()
+
+        def on_evaluate(
+            self,
+            args: TrainingArguments,
+            state: TrainerState,
+            control: TrainerControl,
+            model: PreTrainedModel,
+            **kwargs,
+        ) -> None:
+            self.plo.on_evaluate()  # what to set metric to?
```

Comment:
Another TODO was to check the `TrainerState` of something else holds the current epoch's validation metric.